drivers/backup/portworx/portworx.go

```diff
@@ -68,6 +68,79 @@ type portworx struct {
+			"background":        {"true"},
+			"runInSinglePod":    {"false"},
+			"pod_selector_list": {"app=mysql"},
+			"container":         {"", ""},
+		},
+			"post": {"post_action_list": {"mysql --user=root --password=$MYSQL_ROOT_PASSWORD -Bse 'FLUSH LOGS; UNLOCK TABLES;'"},
+				"background":        {"false"},
+				"pod_selector_list": {"app=mysql"},
+				"runInSinglePod":    {"false"},
+				"container":         {""},
+			},
+		},
+	}
+
+	// AppParametersNew is a map of struct for all the value for rules
+	// This map needs to be updated for new applications as and whe required
+	AppParametersNew = map[string]map[string]backup.RuleStruct{
+		"cassandra": {"pre": backup.RuleStruct{ActionList: []string{"nodetool flush -- keyspace1;", "echo 'test"}, PodSelectorList: []string{"app=cassandra", "app=cassandra1"}, Background: []string{"false", "false"}, RunInSinglePod: []string{"false", "false"}, Container: []string{}},
+			"post": backup.RuleStruct{ActionList: []string{"nodetool verify -- keyspace1;", "nodetool verify -- keyspace1;"}, PodSelectorList: []string{"app=cassandra", "app=cassandra1"}, Background: []string{"false", "false"}, RunInSinglePod: []string{"false", "false"}, Container: []string{}}},
+		"postgres":        {"pre": backup.RuleStruct{ActionList: []string{"PGPASSWORD=$POSTGRES_PASSWORD; psql -U \"$POSTGRES_USER\" -c \"CHECKPOINT\";"}, PodSelectorList: []string{"app=postgres"}, Background: []string{"false"}, RunInSinglePod: []string{"false"}, Container: []string{}}},
+		"postgres-backup": {"pre": backup.RuleStruct{ActionList: []string{"PGPASSWORD=$POSTGRES_PASSWORD; psql -U \"$POSTGRES_USER\" -c \"CHECKPOINT\";"}, PodSelectorList: []string{"app=postgres"}, Background: []string{"false"}, RunInSinglePod: []string{"false"}, Container: []string{}}},
+		"mysql": {"pre": backup.RuleStruct{ActionList: []string{"mysql --user=root --password=$MYSQL_ROOT_PASSWORD -Bse 'FLUSH TABLES WITH READ LOCK;system ${WAIT_CMD};'"}, PodSelectorList: []string{"app=mysql"}, Background: []string{"true"}, RunInSinglePod: []string{"false"}, Container: []string{}},
+			"post": backup.RuleStruct{ActionList: []string{"mysql --user=root --password=$MYSQL_ROOT_PASSWORD -Bse 'FLUSH LOGS; UNLOCK TABLES;'"}, PodSelectorList: []string{"app=mysql"}, Background: []string{"false"}, RunInSinglePod: []string{"false"}, Container: []string{}}},
+		"mysql-backup": {"pre": backup.RuleStruct{ActionList: []string{"mysql --user=root --password=$MYSQL_ROOT_PASSWORD -Bse 'FLUSH TABLES WITH READ LOCK;system ${WAIT_CMD};'"}, PodSelectorList: []string{"app=mysql"}, Background: []string{"true"}, RunInSinglePod: []string{"false"}, Container: []string{}},
+			"post": backup.RuleStruct{ActionList: []string{"mysql --user=root --password=$MYSQL_ROOT_PASSWORD -Bse 'FLUSH LOGS; UNLOCK TABLES;'"}, PodSelectorList: []string{"app=mysql"}, Background: []string{"false"}, RunInSinglePod: []string{"false"}, Container: []string{}}},
+	}
```

Comment:
This whole thing can be even more optimised by declaring our struct in this fashion:

```
type RuleSpec struct {
	ActionList      []string
	PodSelectorList []string
	Background      []string
	RunInSinglePod  []string
	Container       []string
}

type PreRule struct {
	Rule RuleSpec
}

type PostRule struct {
	Rule RuleSpec
}

type AppRule struct {
	PreRule  PreRule
	PostRule PostRule
}
```

By doing this, we can convert the map of map to just a map. Here is a sample of how the AppParams will look:

```
var(
  AppRuleMaster = map[string]backup.AppRule{
		"cassandra": {
			PreRule: backup.PreRule{
				Rule: backup.RuleSpec{
					ActionList: []string{"nodetool flush -- keyspace1;", "echo 'test"}, PodSelectorList: []string{"app=cassandra", "app=cassandra1"}, Background: []string{"false", "false"}, RunInSinglePod: []string{"false", "false"}, Container: []string{},
				},
			},
			PostRule: backup.PostRule{
				Rule: backup.RuleSpec{
					ActionList: []string{"nodetool verify -- keyspace1;", "nodetool verify -- keyspace1;"}, PodSelectorList: []string{"app=cassandra", "app=cassandra1"}, Background: []string{"false", "false"}, RunInSinglePod: []string{"false", "false"}, Container: []string{},
				},
			},
		},
	}

)
```

Reply:
This will look cleaner, well formatted and most importantly we can just check if PreRule is not nil, apply Pre-rule, else don't.

Reply:
Let me know you thoughts

Reply:
done